library/set/divide_spec.rb

```diff
@@ -13,11 +13,16 @@
     ret.sort.should == ["five", "four", "one", "three", "two"]
   end
 
-  # BUG: Does not raise a LocalJumpError, but a NoMethodError
-  #
-  # it "raises a LocalJumpError when not passed a block" do
-  #   lambda { Set[1].divide }.should raise_error(LocalJumpError)
-  # end
+  it "returns an enumerator when not passed a block" do
+    ret = Set[1, 2, 3, 4].divide
+    ret.should be_kind_of(Enumerator)
+    ret.each(&:even?).should == Set[Set[1, 3], Set[2, 4]]
+  end
+
+  it "only uses the first element if the arity is not 2" do
+    set = Set["one", "two", "three", "four", "five"].divide { |x, _, _| x.length }
+    set.map { |x| x.to_a.sort }.sort.should == [["five", "four"], ["one", "two"], ["three"]]
+  end
```

Comment:
**suggestion:** I would move out this case into a separate `describe`/`context' (with a title like "when passed a block with an arity > 2") for consistency.

Reply:
**suggestion:** Also I like the check in the case above:

```ruby
ret.each(&:even?).should == Set[Set[1, 3], Set[2, 4]]
```
using `Set` instead of sorting looks more readable and shorted

Reply:
**suggestion:** TBH I would find a case title a bit laconic and not comprehensive (as well as in a similar case below for arity = 2). It will be great if the titles reflect documentation and describe exactly what happens and how a set elements are compared/divided:

> If the arity of the block is 2, elements o1 and o2 are in common if block.call(o1, o2) is true. Otherwise, elements o1 and o2 are in common if block.call(o1) == block.call(o2).

Reply:
> **suggestion:** I would move out this case into a separate `describe`/`context' (with a title like "when passed a block with an arity > 2") for consistency.

I extracted this one into its own block, and added a block with a splat argument as well (which also should use the first element, since this results in an arity of `-1`, this might be a nice extra test case for Ruby implementations)

I think the other two entries are nice additions, but slightly out of scope for this case.